internal/provider/remote_client.go

```diff
@@ -68,6 +99,18 @@ func (c *RemoteClient) WriteFileShell(content string, path string) error {
 	}
 	defer session.Close()
 
+	cmd := fmt.Sprintf("sudo touch %s", path)
+	err = run(session, cmd)
+	if err != nil {
+		return err
+	}
+
+	cmd = fmt.Sprintf("sudo chmod %s %s", permissions, path)
+	err = run(session, cmd)
```

Comment:
Sorry for the tests not working in CI on fork, but each `run` requires a separete session. I created https://github.com/tenstad/terraform-provider-remote/pull/232 to simplify it a bit, and enable the following. The `session` with `StdinPipe` still remains unchanged.

```suggestion
	cmd := fmt.Sprintf("sudo touch %s", path)
	err = c.run(cmd)
	if err != nil {
		return err
	}

	cmd = fmt.Sprintf("sudo chmod %s %s", permissions, path)
	err = c.run(cmd)
```